docs/guide/migration.md

```diff
@@ -27,27 +27,39 @@ A small fraction of users will now require using [@vitejs/plugin-legacy](https:/
   - `build.polyfillDynamicImport` (use [`@vitejs/plugin-legacy`](https://github.com/vitejs/vite/tree/main/packages/plugin-legacy) for browsers without dynamic import support)
   - `optimizeDeps.keepNames` (switch to [`optimizeDeps.esbuildOptions.keepNames`](../config/dep-optimization-options.md#optimizedepsesbuildoptions))
 
-## Dev Server Changes
+## Optimized deps and SSR changes
+
+These are the biggest architecture changes in Vite v3. To allow projects to migrate from v2 in case of a compat issue, options have been added to revert to the Vite v2 strategies.
+
+:::warning
+These options are marked as experimental and deprecated. They may be removed in a future v3 minor.
+
+- `experimental.devDepsScanner`
+- `experimental.buildRollupPluginCommonjs`
+- `experimental.buildSsrCjsExternalHeuristics`
+  :::
+
+### Dev Server Changes
 
 Vite's default dev server port is now 5173. You can use [`server.port`](../config/server-options.md#server-port) to set it to 3000.
 
 Vite's default dev server host is now `localhost`. You can use [`server.host`](../config/server-options.md#server-host) to set it to `127.0.0.1`.
 
 Vite optimizes dependencies with esbuild to both convert CJS-only deps to ESM and to reduce the number of modules the browser needs to request. In v3, the default strategy to discover and batch dependencies has changed. Vite no longer pre-scans user code with esbuild to get an initial list of dependencies on cold start. Instead, it delays the first dependency optimization run until every imported user module on load is processed.
 
-To get back the v2 strategy, you can use [`optimizeDeps.devScan`](../config/dep-optimization-options.md#optimizedepsdevscan).
+To get back the v2 strategy, you can use [`experimental.devDepsScanner`](../config/dep-optimization-options.md#optimizedepsdevscan).
```

Comment:
Looks like the link here isn't updated (Saw the other was updated), but should we link to a new `experimental.md` file?